hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -180,4 +184,24 @@ public Option<String> getProperty(EngineProperty key) {
   public void setJobStatus(String activeModule, String activityDescription) {
     javaSparkContext.setJobGroup(activeModule, activityDescription);
   }
+
+  @Override
+  public void putCachedDataIds(String basePath, String instantTime, int... ids) {
+    Pair<String, String> key = Pair.of(basePath, instantTime);
+    cachedRddIds.putIfAbsent(key, new ArrayList<>());
```

Comment:
Since we're appending to ArrayList here we need to guard it w/ a lock (and since we'd have to grab lock anyways we can just use HashMap)

Reply:
yup good catch